packages/expo-modules-autolinking/src/fileUtils.ts

```diff
@@ -22,11 +22,14 @@ export async function globMatchFunctorAllAsync(
   matchFunctor: MatchFunctor,
   options?: GlobOptions
 ): Promise<string[]> {
-  const globStream = glob.stream(globPattern, { ...options, withFileTypes: false });
+  const globStream = glob.stream(globPattern, { withFileTypes: true });
   const cwd = options?.cwd !== undefined ? `${options.cwd}` : process.cwd();
   const results: string[] = [];
-  for await (const file of globStream) {
-    let filePath = file.toString();
+  for await (const globPath of globStream) {
+    if (!(globPath as Path).isFile()) {
```

Comment:
I'm pretty sure the `as Path` cast here is redundant and not needed, since the typings of `glob` typically overlay the correct types for `withFileTypes` (as an overload)

Happy to take this over though and push changes as needed, lmk